inhalt/inc.hilfsfunktionen.php

```diff
@@ -1,176 +1,178 @@
-                $retvar = strtr($retvar, array("\x00" => "\\x00", "\x1a" => "\\x1a", "\n" => "\\n", "\r" => "\\r", "\\" => "", "'" => "", "\"" => "")); // entfernt: " ' \
-                return $retvar;
+
+        case 'SQLSAFE':
+            $retvar = stripslashes($value);
+            if ($nl2br) {
+                $retvar = nl2br($retvar);
+            }
+            //$retvar = strtr($retvar, array("\x00" => "\\x00", "\x1a" => "\\x1a", "\n" => "\\n", "\r" => "\\r", "\\" => "\\\\", "'" => "\'", "\"" => "\\\"")); // nur escapen
+            $retvar = strtr($retvar, array("\x00" => '\\x00', "\x1a" => '\\x1a', "\n" => '\\n', "\r" => '\\r', '\\' => '', "'" => '', '"' => '')); // entfernt: " ' \
+            return $retvar;
             break;
-            case 'SHORTNAME':
-                if (!preg_match('/[^0-9A-Za-z_]/',$_GET[$key])) {
-                    return $_GET[$key];
-                }
+
+        case 'SHORTNAME':
+            if (!preg_match('/[^0-9A-Za-z_]/', $value)) {
+                return $value;
+            }
             break;
-            case 'PATHNAME':
-                if (!preg_match('/[^0-9A-Za-z_\/\.:\-]/',$_GET[$key])) {
-                    return $_GET[$key];
-                }
+
+        case 'PATHNAME':
+            if (!preg_match('/[^0-9A-Za-z_\/\.:\-]/', $value)) {
+                return $value;
+            }
             break;
-            default:
-                if (!preg_match('/[^0-9A-Za-z_&:;\-]/',$_GET[$key])) {
-                    return $_GET[$key];
-                }
-        }
+
+        default:
+            if (!preg_match('/[^0-9A-Za-z_&:;\-]/', $value)) {
+                return $value;
+            }
     }
+
     return false;
 }
 
-if (!defined('ONLY_LETTERS')) { define('ONLY_LETTERS',0); }
+if (!defined('ONLY_LETTERS')) { define('ONLY_LETTERS', 0); }
 if (!defined('WITH_NUMBERS')) { define('WITH_NUMBERS', 1); }
 if (!defined('WITH_SPECIAL_CHARACTERS')) { define('WITH_SPECIAL_CHARACTERS', 2); }
 /**
-* Erzeugt einen Zufallsstring
-* 
-* Erzeugt aus Vorgaben einen Zufallsstring
-*@autor finke
-*@return string Zufalsstring
-*/
-function zufallstring($size = 20, $url = ONLY_LETTERS){
-  mt_srand();
-  $pool = 'abcdefghijklmnopqrstuvwxyz';
-  $pool .= 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
-  if($url & WITH_SPECIAL_CHARACTERS){
-    $pool .= ',.-;:_#+*~!§$%&/()=?';
-  }
-  if($url & WITH_NUMBERS){
-    $pool .='0123456789';
-  }
-  $pool = str_shuffle($pool);
-  $pool_size = strlen($pool);
-  $salt ='';
-  for($i = 0;$i<$size; $i++){
-    $salt .= $pool[mt_rand(0, $pool_size - 1)];
-  }
-  return $salt; 
+ * Erzeugt einen Zufallsstring
+ *
+ * Erzeugt aus Vorgaben einen Zufallsstring
+ * @autor finke
+ * @return string Zufalsstring
+ */
+function zufallstring($size = 20, $url = ONLY_LETTERS)
+{
+    mt_srand();
+    $pool = 'abcdefghijklmnopqrstuvwxyz';
+    $pool .= 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+    if ($url & WITH_SPECIAL_CHARACTERS) {
+        $pool .= ',.-;:_#+*~!ยง$%&/()=?';
```

Comment:
Some characters got confused here. I had the same encoding problem which lead to passwords not being recognized when one of those characters was chosen. Maybe its best to remove those two problematic characters completely?